File: VB/WebSite/CustomForms/MyVerticalAppointment.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web.UI.WebControls
Imports System.Web.UI.HtmlControls
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Drawing

Partial Public Class MyVerticalAppointment
	Inherits System.Web.UI.UserControl
	Private ReadOnly Property Container() As VerticalAppointmentTemplateContainer
		Get
			Return CType(Parent, VerticalAppointmentTemplateContainer)
		End Get
	End Property
	Private ReadOnly Property Items() As VerticalAppointmentTemplateItems
		Get
			Return Container.Items
		End Get
	End Property

	Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		appointmentDiv.Style.Value = Items.AppointmentStyle.GetStyleAttributes(Page).Value
		horizontalSeparator.Style.Value = Items.HorizontalSeparator.Style.GetStyleAttributes(Page).Value

		lblStartTime.ControlStyle.MergeWith(Items.StartTimeText.Style)
		lblEndTime.ControlStyle.MergeWith(Items.EndTimeText.Style)
		lblTitle.ControlStyle.MergeWith(Items.Title.Style)
		lblDescription.ControlStyle.MergeWith(Items.Description.Style)

		statusContainer.Controls.Add(Items.StatusControl)
		LayoutAppointmentImages()

		Dim cfPrice As Object = Container.AppointmentViewInfo.Appointment.CustomFields("Price")
		If cfPrice IsNot DBNull.Value Then
			Dim price As Decimal = Convert.ToDecimal(cfPrice)
			lblPrice.Text = price.ToString("c2")

			lblPrice.ControlStyle.MergeWith(Items.Description.Style)
			lblPriceHeader.ControlStyle.MergeWith(Items.Description.Style)
		Else
			lblPrice.Visible = False
			lblPriceHeader.Visible = False
		End If
	End Sub
	Private Sub LayoutAppointmentImages()
		Dim count As Integer = Items.Images.Count
		For i As Integer = 0 To count - 1
			Dim row As New HtmlTableRow()
			Dim cell As New HtmlTableCell()
			AddImage(cell, Items.Images(i))
			row.Cells.Add(cell)
			imageContainer.Rows.Add(row)
		Next i
	End Sub
	Private Sub AddImage(ByVal targetCell As HtmlTableCell, ByVal imageItem As AppointmentImageItem)
		Dim image As New Image()
		imageItem.ImageProperties.AssignToControl(image, False)
		targetCell.Controls.Add(image)
	End Sub
End Class

File: CS/WebSite/CustomForms/MyVerticalAppointment.ascx.cs
using System;
using System.Web.UI.WebControls;
using System.Web.UI.HtmlControls;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Drawing;

public partial class MyVerticalAppointment : System.Web.UI.UserControl {
	VerticalAppointmentTemplateContainer Container { get { return (VerticalAppointmentTemplateContainer)Parent; } }
	VerticalAppointmentTemplateItems Items { get { return Container.Items; } }

	protected void Page_Load(object sender, EventArgs e) {
		appointmentDiv.Style.Value = Items.AppointmentStyle.GetStyleAttributes(Page).Value;
		horizontalSeparator.Style.Value = Items.HorizontalSeparator.Style.GetStyleAttributes(Page).Value;

		lblStartTime.ControlStyle.MergeWith(Items.StartTimeText.Style);
		lblEndTime.ControlStyle.MergeWith(Items.EndTimeText.Style);
		lblTitle.ControlStyle.MergeWith(Items.Title.Style);
		lblDescription.ControlStyle.MergeWith(Items.Description.Style);
				
		statusContainer.Controls.Add(Items.StatusControl);
		LayoutAppointmentImages();
		
		object cfPrice = Container.AppointmentViewInfo.Appointment.CustomFields["Price"];
		if (cfPrice != DBNull.Value) {
			Decimal price = Convert.ToDecimal(cfPrice);
			lblPrice.Text = price.ToString("c2");

			lblPrice.ControlStyle.MergeWith(Items.Description.Style);
			lblPriceHeader.ControlStyle.MergeWith(Items.Description.Style);
		}
		else {
			lblPrice.Visible = false;
			lblPriceHeader.Visible = false;
		}		
	}
	void LayoutAppointmentImages() {
		int count = Items.Images.Count;
		for (int i = 0; i < count; i++) {
			HtmlTableRow row = new HtmlTableRow();
			HtmlTableCell cell = new HtmlTableCell();
			AddImage(cell, Items.Images[i]);
			row.Cells.Add(cell);
			imageContainer.Rows.Add(row);
		}
	}
	void AddImage(HtmlTableCell targetCell, AppointmentImageItem imageItem) {
		Image image = new Image();
		imageItem.ImageProperties.AssignToControl(image, false);
		targetCell.Controls.Add(image);
	}	
}